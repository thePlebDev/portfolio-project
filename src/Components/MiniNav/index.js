import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Container = styled.div`
    display:flex;
    justify-content:center;
    align-items:center;
    margin-top:20px;
`
const Text = styled.span`
padding:10px;
font-size:25px;
text-transform:uppercase;
opacity:0.6;
cursor:pointer;
position:relative;

&:after{
  content:'';
  background-color:red;
  height:2px;
  width:0px;
  position:absolute;
  bottom:0;
  left:0;
  right:0;
  margin:0 auto;
  transition:all .3s;

};

&:hover:after{
  width:90px;
}
`




const MiniNav =()=>{

  return(
    <Container>
      <Link to="/">
        <Text>Home</Text>
      </Link>
      <Link to="/blog">
        <Text>Blog</Text>
      </Link>
    </Container>
  )
}

export default MiniNav
