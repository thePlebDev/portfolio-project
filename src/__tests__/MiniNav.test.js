import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import {BrowserRouter as Router} from 'react-router-dom'

import MiniNav from '../Components/MiniNav'

const setup = (props={})=>{
  const component = shallow(<MiniNav {...props} />)
  return component
}


describe('testing the MiniNav component',()=>{
  it('should test the snapshot',()=>{
    const tree = renderer.create(<Router><MiniNav /></Router>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render normaly',()=>{
    const wrapper = setup(<MiniNav/>)
    expect(wrapper.find('[data-testid="home"]').length).toBe(1)
    expect(wrapper.find('[data-testid="blog"]').length).toBe(1)


  })
})
