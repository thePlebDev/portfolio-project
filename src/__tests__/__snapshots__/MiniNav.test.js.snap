// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the MiniNav component should test the snapshot 1`] = `
<div
  className="sc-bdnylx dnqPyt"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <span
      className="sc-gtssRu bZyBwx"
      data-testid="home"
    >
      Home
    </span>
  </a>
  <a
    href="/blog"
    onClick={[Function]}
  >
    <span
      className="sc-gtssRu bZyBwx"
      data-testid="blog"
    >
      Blog
    </span>
  </a>
</div>
`;
